### PR TITLE
Add cross-compile build for FreeBSD

### LINKED ISSUE #9465: Add FreeBSD build to CI via cross-compilation
https://linux-tips.com/t/how-to-build-a-freebsd-toolchain-for-linux/288

### PR DESCRIPTION
**Pull request by abyss7**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add cross-compile build for FreeBSD. This fixes #9465

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
> 00834_kill_mutation

- flacky test.

File: dbms/CMakeLists.txt [hunk 1]
```diff
@@ -258,9 +258,9 @@ set_source_files_properties(src/Common/malloc.cpp PROPERTIES COMPILE_FLAGS "-fno
 add_library (clickhouse_new_delete STATIC src/Common/new_delete.cpp)
 target_link_libraries (clickhouse_new_delete PRIVATE clickhouse_common_io)
 
-if (OS_FREEBSD)
-    target_compile_definitions (clickhouse_common_io PUBLIC CLOCK_MONOTONIC_COARSE=CLOCK_MONOTONIC_FAST)
-endif ()
+#if (OS_FREEBSD)
+#    target_compile_definitions (clickhouse_common_io PUBLIC CLOCK_MONOTONIC_COARSE=CLOCK_MONOTONIC_FAST)
```
Comment (alexey-milovidov) [inline dbms/CMakeLists.txt]: Why not? I'm afraid that CLOCK_MONOTONIC_COARSE is used in places where CLOCK_MONOTONIC may appear to be too slow.

File: dbms/src/Common/setThreadName.cpp [hunk 2]
```diff
@@ -29,11 +29,10 @@ void setThreadName(const char * name)
         throw DB::Exception("Thread name cannot be longer than 15 bytes", DB::ErrorCodes::PTHREAD_ERROR);
 #endif
 
-#if defined(__FreeBSD__)
+#if defined(OS_FREEBSD)
     pthread_set_name_np(pthread_self(), name);
-    return;
-
-#elif defined(__APPLE__)
+    if ((false))
```
Comment (alexey-milovidov) [inline dbms/src/Common/setThreadName.cpp]: Looks strange.

**Review by alexey-milovidov — APPROVED** [review]

File: dbms/CMakeLists.txt (hunk 1 above)
Comment (abyss7) [reply dbms/CMakeLists.txt]: I put it in <ext/time.h>

File: dbms/src/Common/setThreadName.cpp (hunk 2 above)
Comment (abyss7) [reply dbms/src/Common/setThreadName.cpp]: Clang issues a warning with this specific workaround for explicitly dead code.

**Comment by hellvesper** [comment]
This is great, is there build instruction? I'm trying to build on Ubuntu 18.04 with clan8 and 9 without luck.

**Pull request merged**